docs/test-architecture.md

```diff
@@ -0,0 +1,61 @@
+Testing via Helm
+================
+
+Problem Summary
+---------------
+
+Currently in helm/charts we have a simple way of health-checking charts which
+checks that all pods in the chart reach "Running" state on a running Kubernetes
+cluster.In order to build a system that holds chart quality at the forefront,
+we need to introduce an easy way to create and run chart-centric tests.
+
+Proposed Solution
+-----------------
+
+#### User Experience
+
+```
+# this UX stays the same
+helm update
+helm fetch deis/deis
+helm install deis
```

Comment:
Likewise, instead of fetching a chart to the client and installing to a cluster from there, we should now be telling the helm (formerly dm) server in the cluster to install it.